packages/grafana-data/src/transformations/transformers/formatString.ts

```diff
@@ -0,0 +1,90 @@
+import { map } from 'rxjs/operators';
+
+import { Field, FieldType } from '../../types';
+import { DataTransformerInfo } from '../../types/transformations';
+
+import { DataTransformerID } from './ids';
+
+export enum FormatStringOutput {
+  UpperCase = 'Upper Case',
+  LowerCase = 'Lower Case',
+  FirstLetter = 'Capitalize First Letter',
+  EveryFirstLetter = 'Capitalize First Letter Of Every Word',
+  PascalCase = 'PascalCase',
+  CamelCase = 'camelCase',
+}
+
+export interface FormatStringTransformerOptions {
+  stringField: string;
+  outputFormat: string;
+}
+
+export const formatStringTransformer: DataTransformerInfo<FormatStringTransformerOptions> = {
+  id: DataTransformerID.formatString,
+  name: 'Format string',
+  description: 'Set the capitalisation of a string field',
+  defaultOptions: { stringField: '', outputFormat: '' },
+  operator: (options) => (source) =>
+    source.pipe(
+      map((data) => {
+        const formatter = createStringFormatter(options.stringField, options.outputFormat);
+
+        if (!Array.isArray(data) || data.length === 0) {
```

Comment:
`!Array.isArray(data)` - is it possible that the `data` here is not an array?

Reply:
Hrm, I don't think so? I've noticed this line in other Transformers (specifically in this case I was using `formatTime` as a base to sample from), but I believe you're right that it's not strictly neccesary.